File: microservicio-clinica-nest/src/modules/clinical-records/dto/create-clinical-record-in.dto.ts
import { IsInt, IsNotEmpty, IsString, IsOptional, IsDateString, Min } from 'class-validator';
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export class CreateClinicalRecordInDto {

@ApiProperty({ 
    example: 1,
    description: 'ID del paciente',
  })
  @IsInt({ message: 'El ID del paciente debe ser un número entero' })
  @Min(1, { message: 'El ID del paciente debe ser mayor a 0' })
  @IsNotEmpty({ message: 'El ID del paciente es obligatorio' })
  patientId: number;

  @ApiProperty({ 
    example: 1,
    description: 'ID del tipo de tumor',
  })
  @IsInt({ message: 'El ID del tipo de tumor debe ser un número entero' })
  @Min(1, { message: 'El ID del tipo de tumor debe ser mayor a 0' })
  @IsNotEmpty({ message: 'El ID del tipo de tumor es obligatorio' })
  tumorTypeId: number;

  @ApiProperty({ 
    example: '2023-05-20',
    description: 'Fecha de diagnóstico (formato YYYY-MM-DD)',
  })
  @IsDateString({}, { message: 'La fecha debe tener formato YYYY-MM-DD' })
  @IsNotEmpty({ message: 'La fecha de diagnóstico es obligatoria' })
  diagnosisDate: string;

  @ApiPropertyOptional({ 
    example: 'IIA',
    description: 'Etapa del cáncer (ej: IIA, III, IV)',
  })
  @IsString({ message: 'La etapa debe ser texto' })
  @IsOptional()
  stage?: string;

  @ApiPropertyOptional({ 
    example: 'Cirugía conservadora + quimioterapia adyuvante con docetaxel',
    description: 'Protocolo de tratamiento',
  })
  @IsString({ message: 'El protocolo de tratamiento debe ser texto' })
  @IsOptional()
  treatmentProtocol?: string;

}

File: autenticación/src/main/java/com/creators/autenticacion/models/dto/clinicalRecord/CreateClinicalRecordInDto.java
package com.creators.autenticacion.models.dto.clinicalRecord;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;

@Data
@Schema(description = "DTO para crear una nueva historia clínica")

public class CreateClinicalRecordInDto {

    @Schema(description = "ID del paciente", example = "1", required = true)
    private Integer patientId;

    @Schema(description = "ID del tipo de tumor", example = "1", required = true)
    private Integer tumorTypeId;

    @Schema(description = "Fecha de diagnóstico (formato YYYY-MM-DD)", example = "2023-05-20", required = true)
    private String diagnosisDate;

    @Schema(description = "Etapa del cáncer (ej: IIA, III, IV)", example = "IIA")
    private String stage;

    @Schema(description = "Protocolo de tratamiento", example = "Cirugía conservadora + quimioterapia adyuvante con docetaxel")
    private String treatmentProtocol;

}
